code/src/components/Form.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+
+export const Form = () => {
+  const [newThought, setNewThought] = useState("");
+  const POST_URL = "https://happy-thoughts-technigo.herokuapp.com/thoughts";
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    //Creates a POST request to add new Thought to the API
+    fetch(POST_URL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      //Adds new thought to the API's body in json format
+      //newThought is the value we got from the input field
+      body: JSON.stringify({ message: newThought }),
+    }).then(() => {
+      //Asks page to reload after the new thought has been POSTed
+      //So the click should POST the thought to the API then refresh
+      //the page and the new page should now also show the thought we
+      //just added
+      window.location.reload();
+    });
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="thoughts-form">
+      <p className="form-title">What's making you happy right now?</p>
+      <textarea
+        type="text"
+        maxLength="140"
+        onChange={(event) => setNewThought(event.target.value)}>
+      </textarea>
+      <p
+        className="text-counter"
+        //Condition to change text color to red when there are only 10 characters left to reach max
+        style={{ color: newThought.length > 130 ? "red" : "black" }}>
+        {140 - newThought.length} characters left
+      </p>
+      <button
+        className="form-button"
+        type="submit"
+        /* Condition to enable the submit button only when the thought has right amount of characters */
+        disabled={
+          newThought.length < 6 || newThought.length > 140 ? true : false
```

Comment:
When we want to set attribute to a boolean value, we can omit the part about returning `false` or `true`.

This is overcomplicated piece of code :

```jsx
<button
  className="button"
  disabled={message.length < 6 || message.length > 140 ? true : false}
>
  Click me!
</button>
```

This is shorter, more accurate version of the same code :

```jsx
<button
  className="button"
  disabled={message.length < 6 || message.length > 140}
>
  Click me!
</button>
```

It does work, because logical operations such as && or || are always processed into boolean values at the end :

```jsx
const message = 'I am a string longer than 6, but shorter than 140'

message.length < 6 // This line of code is processed into false, because length
                   // of message is higher than 6

message.length > 140 // This line of code is also processed into false, because
                     // length of message is lower than 140

// This line of code
message.length < 6 || message.length > 140

// is basically
false || false

// therefore it's
false

// Final result of
disabled={message.length < 6 || message.length > 140}

// is
disabled={false || false}

// so
disabled={false}
```